### ISSUE TITLE
[Feature request] Reverting a pull request

### ISSUE THREAD

**Issue by ghost** [go-gitea/gitea#6375]
Please consider adding a feature to revert pull requests, just like on github: https://help.github.com/en/articles/reverting-a-pull-request

**Comment by fusen245** [comment]
komentarz ze to bedzie bardzo useful czy cos

**Comment by 6543** [comment]
@lunny can you add this to v1.11.x ?

because it sounds very usefull

**Comment by lunny** [comment]
@6543 but nobody are working on this even I think it's not difficult to implement.

**Comment by vjgarcera** [comment]
Is there any plan to implement this feature? Thanks.

**Comment by 6543** [comment]
@vjgarcera plan: well issue is till open ;)

But ine dev has picked it jet, feel free to have a look

**Comment by wxiaoguang** [comment]
A commit could be reverted on the web UI, so reverting the PR's commit could revert that PR:

![image](https://github.com/user-attachments/assets/2e5c8397-9f86-4490-a70c-599cb8e6b9e4)

**Comment by lunny** [comment]
Yes, so we just need a button in the merged Pull Request View which can be linked to the same URL. It should not be a very big change from current codebase.

**Comment by germanne** [comment]
I would be also very interested in this feature. Also the goal is not to get a revert PR but to quickly put the branch back to its state before merging, which can be very useful in some cases of problems

**Comment by lunny** [comment]
If you mean reverting a commit, that has been implemented for a long time.

![Image](https://github.com/user-attachments/assets/d6a9e63f-5ac1-4c05-8636-5c6e345c4080)

**Comment by germanne** [comment]
that's exactly my point, I'd like to be able to do that on the MR and viewout any "revert commit"! So reverting without any trace of the merge

**Comment by wxiaoguang** [comment]
> that's exactly my point, I'd like to be able to do that on the MR and viewout any "revert commit"! So reverting without any trace of the merge

That's hard-reset + force-push, which is an unacceptable operation in most development environments, especially when it is done on the "main" branch. I do not think Gitea should provide such support. If you really need to "revert a commit without any trace", you need to do that by your git client and tell all your developers what happens.

**Comment by germanne** [comment]
I thought this was available on Gitlab and indeed just check and it's not, so ok my bad!

**Comment by lunny** [comment]
I think this operation should be try to create a revert pull request. And of course, like @wxiaoguang said, sometimes it will fail.
